### PR TITLE
Add foundation and design tokens

### PR DESCRIPTION
**Pull request by okonet**
2018-04-04 at 16 02

### CODE REVIEW

File: docs/Foundation.md [hunk 1]
```diff
@@ -1 +1,6 @@
-TODO
+Colors, spacing, and typography are the building blocks of the design system. Those values are 
+provided to the application via `ThemeProvider` and the `theme` object. Every element of the 
+design system should be in relation to those values in order to be consistent.
+
+So, intead of writing `color: #8667a8` you should use `theme` prop `color: ${props => props.theme
```
Comment (sapegin) [inline docs/Foundation.md]: I'd show a real code example here, like:

```diff
const Button = style.button`
-   color: #8667a8
+   color: ${props => props.theme.color.primary}
`
```

File: src/theme.js [hunk 2]
```diff
@@ -22,10 +22,11 @@ export default {
 		light: '#ccc',
 		link: '#8667a8',
 		hover: '#ed9dc5',
-		focus: '#ed9dc599',
```
Comment (sapegin) [inline src/theme.js]: Any reason for this change?

File: src/components/foundation/Spacing.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const SpacingExample = styled(Card)`
+	padding: 0;
+	height: ${props => props.space}px;
+	margin: 0;
+	font-size: ${props => props.fontSize};
+	font-family: ${props => props.theme.fontFamily.base};
+	background: ${props => props.theme.color.light};
+	border: none;
+`;
+const SpaceSample = ({ space, children }) => <Box m={0}>
```
Comment (sapegin) [inline src/components/foundation/Spacing.js]: I'd add some space between samples, otherwise it looks like 0 is a label for a 4px bar.

**Review by sapegin — COMMENTED** [review]
Cool!

File: src/theme.js (hunk 2 above)
Comment (okonet) [reply src/theme.js]: Polished was not able to parse it and so does IDEA. I'm not sure how RGBA support looks like but I'd rather use `rgba` or `polished` `transparentize` here.

Comment (sapegin) [reply src/theme.js]: OK, let's use rgba then ;-)

Comment (okonet) [reply src/theme.js]: The support for `#rrggbbaa` is still limited: https://caniuse.com/#feat=css-rrggbbaa so I'd opt for using `rgba` or polished helpers here.

Saying that, I've submitted an issue: https://github.com/styled-components/polished/issues/306 and might work on the fix.

**Review by sapegin — APPROVED** [review]

**Comment by sapegin** [comment]
Looks good now! Feel free to merge if you're not going to do anything else here.

**Pull request merged**